### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by neenjaw**
I have some organizational changes to the codebase planned to be capable to better integrate new types of macro tests.  I don't think I will be finishing the new types of tests in the next few days, but these changes make it easier to modularly expand the analyzer.

Notable changes:
- Exercise Analysis test suites moved from a deep sub-dir to `lib/test_suite/___.ex`
- Feature tests (the current analyzer test) refactored to separate module, imported (via `use`) by the existing `ExerciseTest`

---

Some intermediate commits reference the `ensure_call` test type, that is not included in this PR.

### CODE REVIEW

File: config/exercise_data.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+    "code_file": "two_fer.ex",
+    "analyzer_module": "ElixirAnalyzer.TestSuite.TwoFer"
+  },
+  "pacman-rules": {
+    "code_file": "pacman_rules.ex",
+    "analyzer_module": "ElixirAnalyzer.TestSuite.PacmanRules"
+  },
+  "take-a-number": {
+    "code_file": "take_a_number.ex",
+    "analyzer_module": "ElixirAnalyzer.TestSuite.TakeANumber"
```
Comment (angelikatyborska) [inline config/exercise_data.json:12]: The same renaming needs to happen in files in `doc/`.

Comment (neenjaw) [reply config/exercise_data.json:12]: This should be addressed now

File: docs/step-04/step-04.md [hunk 2]
```diff
@@ -2,33 +2,34 @@
...
-          def hello(name) do
-            "Hello, #{name}!"
-          end
-        end
-      end
-    end
-    ```
+   ```elixir
+   # 04-example-analysis-test.exs
+   defmodule ElixirAnalyzer.ExerciseTesTestSuitet.ExampleTest do
```
Comment (angelikatyborska) [inline docs/step-04/step-04.md:18]: Something went wrong here :D
```suggestion
   defmodule ElixirAnalyzer.TestSuite.ExampleTest do
```

**Review by angelikatyborska — APPROVED** [review]

File: docs/step-04/step-04.md (hunk 2 above)
Comment (neenjaw) [reply docs/step-04/step-04.md:18]: 😱 Thanks!

**Pull request merged**